### PR TITLE
Added asarray method to convert regular-grid data

### PR DESCRIPTION
**Pull request by mauro3**
Fixes https://github.com/JuliaEarth/GeoStats.jl/issues/147.  Let me know if it needs more (maybe docs?).

### CODE REVIEW

File: src/data.jl
Comment (juliohm) [inline src/data.jl:127]: I wonder if we should also provide a method for the general domain case. This method would simply return a flat vector without reshaping, makes sense?

Other than that, please use the same identifiers in the other implementations, i.e. `sdata` for spatial data and `var` for spatial variable. The docstring could say something like "Returns the variable `var` as a Julia array with size equal to the size of the underlying domain." and then provide an example with regular grids.

Also, I wonder if this should be part of the public interface for `AbstractData` more generally, and not just the particular instance `SpatialData` implemented in the package.

File: test/data.jl
Comment (juliohm) [inline test/data.jl:4]: can you please rename this to `z`

**Review by juliohm — CHANGES_REQUESTED** [review]
Thank you for the pull request 💯 please consider the minor adjustments in the review.

File: src/data.jl
Comment (mauro3) [reply src/data.jl:127]: I don't really have an understanding of the datatypes.  So, you'll have to be specific on what types I have to add the method to.

Comment (juliohm) [reply src/data.jl:127]: We have `AbstractData` to represent any spatial data handled by the project. In particular, we provide two concrete implementations `SpatialData` and `DataView` (should probably be renamed to `GeoData` and `GeoDataView` in the future).

So we could provide a fallback implementation to the abstract parent type:

```julia
asarray(sdata::AbstractData, var) = sdata[var]
```

that returns flat vectors, and then specialize for regular grids as you did with:

```julia
asarray(sdata::SpatialData{<:RegularGrid}, var) = reshape(sdata[var], size(domain(sdata)))
```

The `DataView` case will call the fallback as there is no guarantee that a view preserves regularity of samples. At least that is how I see this `asarray` functionality currently.

**Comment by juliohm** [comment]
@mauro3 did you have a chance to take a look into this? I think we can generalize this even further and just provide a single method for the abstract case as follows:

```julia
function asarray(sdata::AbstractData, var::Symbol)
  D = domain(sdata)
  D isa RegularGrid ? reshape(sdata[var], size(D)) : sdata[var]
end
```

That way this function will be part of the public api and we will never need to implement it for derived concrete types.

File: src/data.jl
Comment (mauro3) [reply src/data.jl:127]: Regarding: "We have AbstractData to represent any spatial data handled by the project. In particular, we provide two concrete implementations SpatialData and DataView (should probably be renamed to GeoData and GeoDataView in the future)."

I do use GeoData.jl together with GeoStats.jl, so that rename would be a bit inconvenient, just my 2c.

Comment (juliohm) [reply src/data.jl:127]: I guess the data type in GeoData.jl is actually `GeoArray` right? I need to go over the entire GeoStats.jl stack to make these names more consistent with a `Geo` prefix. AFAIU, Julia can differentiate the module `GeoData` from the data type `GeoStats.GeoData`, that shouldn't be an issue. Please correct me if I am wrong.

Comment (mauro3) [reply src/data.jl]: Consider
```
julia> abstract type GeoStatsBase end

julia> using GeoStatsBase
WARNING: using GeoStatsBase.GeoStatsBase in module Main conflicts with an existing identifier.
ERROR: importing GeoStatsBase into Main conflicts with an existing identifier
```
which would happen analogously if GeoStats.jl was imported before GeoData.jl.  If the other way round, then GeoData.jl would shadow GeoStatsBase.GeoArray.  However, with Julia 1.6 packages can be renamed on import: https://github.com/JuliaLang/julia/blob/master/HISTORY.md

Yes, the datatype is `GeoArray`.

**Comment by mauro3** [comment]
Thanks for the help!  I also added a few tests for the other code-path.

File: src/data.jl
Comment (juliohm) [reply src/data.jl]: The real underlying issue is that GeoData.jl should be named GeoArrays.jl as it defines array types, and not general geospatial data. I will think about this prefix issue, perhaps `Spatial` is enough, but I also dislike the word `Spatial` because it is not the same as `Geospatial`.

File: src/data.jl
Comment (juliohm) [inline src/data.jl:39]:
```suggestion
"""
    asarray(sdata, var)

Returns the data for the variable `var` in spatial data `sdata` as a Julia
array with size equal to the size of the underlying domain if the size is
defined, otherwise returns a vector.
"""
```

**Review by juliohm — CHANGES_REQUESTED** [review]
Awesome! Thanks for the contribution, just a minor suggestion before we can merge ❤️

**Comment by mauro3** [comment]
Cool.  Probably you want to squash before the merge.

**Comment by juliohm** [comment]
Will just wait for the tests, then square and merge :)

**Review by juliohm — APPROVED** [review]

**Pull request merged**
